### PR TITLE
eth/tracers: improve tracing performance (#23016)

### PR DESCRIPTION
**Pull request by Mercybudda**
Improves the performance of debug.traceTransaction
Description
Cherry-pick the commit https://github.com/ethereum/go-ethereum/pull/23016
### Preflight checks

- [x] build passed (`make build`)
- [ ] tests passed (`make test`)

### CODE REVIEW

**Review by guagualvcha — APPROVED** [review]

File: core/vm/gen_structlog.go [hunk 1]
```diff
@@ -4,11 +4,11 @@ package vm
 
 import (
 	"encoding/json"
-	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/holiman/uint256"
```
Comment (yutianwu) [inline core/vm/gen_structlog.go:11]: fix imports

File: core/vm/logger.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ import (
 	"github.com/ethereum/go-ethereum/common/math"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
```
Comment (yutianwu) [inline core/vm/logger.go:32]: same here

File: core/vm/gen_structlog.go (hunk 1 above)
Comment (Mercybudda) [reply core/vm/gen_structlog.go]: Done

File: core/vm/logger.go (hunk 2 above)
Comment (Mercybudda) [reply core/vm/logger.go]: Done

**Review by yutianwu — APPROVED** [review]

**Pull request merged**